File: Daily Challenge/623. Add One Row to Tree.cs
public class Solution 
{
    public TreeNode AddOneRow(TreeNode root, int val, int depth) 
    {
        if (depth == 1) return new TreeNode(val, root, null);
        
        Queue<TreeNode> queue = new();
        queue.Enqueue(root);
        
        int currentDepth = 1;
        while (currentDepth < depth - 1)
        {
            int levelSize = queue.Count;
            for (int i = 0; i < levelSize; i++)
            {
                TreeNode node = queue.Dequeue();
                
                if (node.left is not null) queue.Enqueue(node.left);
                if (node.right is not null) queue.Enqueue(node.right);
            }
            currentDepth++;
        }
        
        while (queue.Count > 0)
        {
            TreeNode node = queue.Dequeue();
            
            node.left = new TreeNode(val, node.left, null);
            node.right = new TreeNode(val, null, node.right);
        }
        
        return root;
    }
}

File: Daily Challenge/623. Add One Row to Tree.py
class Solution:
    def addOneRow(self, root: TreeNode, val: int, depth: int) -> TreeNode:
        return self.Helper(root, val, depth, 1)

    def Helper(self, node: TreeNode, value: int, depth: int, currDepth: int) -> TreeNode:
        if depth == 1:
            return TreeNode(value, node, None)

        if node is None:
            return node

        if currDepth == depth - 1:
            leftNode = node.left
            rightNode = node.right

            node.left = TreeNode(value, None, None)
            node.right = TreeNode(value, None, None)

            node.left.left = leftNode
            node.right.right = rightNode
            return node

        node.left = self.Helper(node.left, value, depth, currDepth + 1)
        node.right = self.Helper(node.right, value, depth, currDepth + 1)

        return node
        